packages/components/src/modal/style.scss

```diff
@@ -5,53 +5,44 @@
 	right: 0;
 	bottom: 0;
 	left: 0;
-	background-color: rgba($black, 0.7);
+	padding: 20px;
+	background: $white;
 	z-index: z-index(".components-modal__screen-overlay");
+	overflow: auto;
+
+	@include break-small() {
+		padding: 100px;
+	}
+}
 
-	// This animates the appearance of the white background.
-	@include edit-post__fade-in-animation();
+// These are two divs rendered by Higher-order components
+// We need their height to be 100% for the verticalcentering
+// of the modal.
+.components-modal__screen-overlay > div,
+.components-modal__screen-overlay > div > div {
+	height: 100%;
 }
 
 // The modal window element.
 .components-modal__frame {
-	// On small screens the content needs to be full width because of limited
-	// space.
-	position: absolute;
-	top: 0;
-	right: 0;
-	bottom: 0;
-	left: 0;
-	box-sizing: border-box;
-	margin: 0;
-	border: $border-width solid $light-gray-500;
-	background: $white;
-	box-shadow: $shadow-modal;
-	overflow: auto;
-
-	// Show a centered modal on bigger screens.
-	@include break-small() {
-		top: 50%;
-		right: auto;
-		bottom: auto;
-		left: 50%;
-		min-width: $modal-min-width;
-		max-width: calc(100% - #{ $grid-size-large } - #{ $grid-size-large });
-		max-height: calc(100% - #{ $header-height } - #{ $header-height });
-		transform: translate(-50%, -50%);
+	max-width: 800px;
+	min-height: 100%;
+	margin: auto;
+	display: grid;
+	align-items: center;
 
-		// Animate the modal frame/contents appearing on the page.
-		animation: components-modal__appear-animation 0.1s ease-out;
-		animation-fill-mode: forwards;
-		@include reduce-motion("animation");
-	}
+	// Animate the modal frame/contents appearing on the page.
+	transform-origin: top center;
+	animation: components-modal__appear-animation 0.2s forwards;
```

Comment:
the animation length here is quite small, maybe `0.4s` can be good, it should however be coupled with animation on `.components-modal__screen-overlay` or else it will be useless.